Update HRA user's guide for the recent rewrite

phargogh opened the pull request:
This PR features the latest changes to the HRA model, including

* An updated outputs and intermediate outputs section
* Much better-specified math (I believe everything important is reflected in there now)

And another of Rafa's files got added in when I did a couple merges to get the latest changes into this branch, so I hope I haven't messed anything up there.

dcdenu4 commented on source/habitat_risk_assessment.rst:
```diff
@@ -82,11 +82,11 @@ The risk of human activities to habitats or species is modeled in five steps.
 where :math:`E_{jkl}` is the exposure score specific to habitat *j*, from stressor *k* in location *l*; :math:`C_{jkl}` is the consequence score, :math:`e_{ijkl}` is the exposure rating criterion *i*, specific to habitat *j* and stressor *k* and location *l*; :math:`c_{ijkl}` is the consequence rating. :math:`d_{ijkl}` represents the data quality rating, :math:`w_{ijkl}` represents the importance ...
 
-**Step 2.** The second step combines the exposure and response values to produce a risk value for each stressor-habitat combination in each grid cell. There are two options for risk calculation.
+**Step 2.** The second step combines the exposure and response values to produce a distance-weighted risk value for each stressor-habitat combination in each grid cell. There are two options for risk calculation and 3 options for distance-based weighting.
```
Should we be consistent with using numbers vs spelling out.
>There are **two** options for risk calculations and **3** options for...

phargogh replied:
Yes we should!  26a056b

dcdenu4 commented on source/habitat_risk_assessment.rst:
```diff
@@ -108,17 +145,68 @@ For Multiplicative Risk calculation, risk to habitat *j* caused by stressor *k*
+* :math:`r_{max}` is the user-defined maximum score.
+
+
+**Step 4.** Each grid cell for each habitat or species is classified as LOW
+(1), MED (2) or HIGH (3) risk based on risk posed by the cumulative effects of
+multiple stressors.
```
I might not be reading this right, but don't we have a Step 4. already? Is leftover from the previous version?

phargogh replied:
We sure do.  I've updated this in 7828994 in an attempt to better express how classification goes by doing the following:

* Reworking the steps 4 and 5 into 4, 4a and 4b to better reflect the interrelation of the classifications.
* The equations now more correctly (I think) capture the `max` part of `4b`.
* The equations in step 4 subsection 1 and step 4 subsection 2 are also linked to the related equations in 4a and 4b, respectively.

dcdenu4 commented on source/habitat_risk_assessment.rst:
```diff
@@ -446,6 +536,29 @@ Preparing Spatially Explicit Criteria Layers
+Defining the Number of Overlapping Stressors
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The number of overlapping stressors is used in determining breaks between high, medium and low risk classifications.
```
```suggestion
The number of overlapping stressors is used in determining breaks between high, medium, and low risk classifications.
```

dcdenu4 commented on source/habitat_risk_assessment.rst:
```diff
@@ -460,110 +573,149 @@ Output Folder
+  This raster layer depicts the sum of habitat cumulative risk scores divided
+  by the number of habitats occurring in each cell. It is best interpreted as
+  an average risk across all habitats in a grid cell. For example, in a
+  nearshore grid cell that contains some coral reef, mangrove and soft bottom
```
```suggestion
  nearshore grid cell that contains some coral reef, mangrove, and soft bottom
```

dcdenu4 commented on source/habitat_risk_assessment.rst:
```diff
@@ -460,110 +573,149 @@ Output Folder
++ **RECOVERY_<habitat>.tif**
+  A raster file depicting the resilience or recovery potential for the given
+  habitat or species for each cell. Recovery potential is based on natural
+  mortality rate, recruitment rate, age at maturity/recovery time and
```
```suggestion
  mortality rate, recruitment rate, age at maturity/recovery time, and
```

dcdenu4 reviewed (changes requested):
Hey @phargogh , sorry for not getting to this sooner. I just had a few minor comments / suggestions. It looks like there is a conflict to work through around Rafa's files as well...

Thanks!

phargogh commented:
Thanks @dcdenu4!  I believe I've addressed all of your comments.  The main change I wanted to mention is the slight restructuring of the step 4 into step 4, 4a and 4b, which I hope is clearer.  Let me know what you think!

dcdenu4 reviewed (commented):
Hey @phargogh, I'm trying build a local version to just give it a lookover, but of course having issues with my environment. I just had a comment about the `requirements.txt` update.

I also just wanted to make sure the addition of Rafa's image isn't overwriting anything Stacie recently swapped out.

So close to getting this UG release stuff out!

phargogh commented on requirements.txt:
```diff
@@ -1,5 +1,6 @@
 git-lfs
-natcap.invest
+#natcap.invest
+git+https://github.com/natcap/invest.git@release/3.12#egg=natcap.invest
```
> Is this part of our normal workflow for releases?

I would argue yes, even though we haven't done this kind of thing for some time and it hasn't been necessary in the UG until now because of the UG's relatively recent dependency on `natcap.invest`.

phargogh commented:
@dcdenu4 I just merged in the latest changes I've made on `release/3.12` to correct the build failure, so maybe that'll also address the environment issue on your local machine?

I removed the JPG that somehow got added ... the SDR UG chapter uses a PNG nowadays, so I'm guessing the JPG got added through a merge somehow.

Other than that (and assuming the builds are good now), this might be ready!

phargogh commented:
@dcdenu4 sorry about that!  I was referring to equations using `:ref:<equation label>` instead of the correct `:eq:<equation label>`.  This has been corrected and our GHA builds are now passing again.
